current/runtime/fees.md

```diff
@@ -4,118 +4,151 @@ lang: en
 
-### Inclusion Fee
+### Mandatory Dispatches
 
-A transaction fee consists of three parts:
+The most critical dispatches are classified as mandatory and represent dispatches that are part of
+the block validation process; they must be inherent, meaning they may not be signed. Mandatory
+dispatches will be included even if they cause a block to surpass the weight limit, so it is vital
+to ensure that they are validated separately. This can typically be accomplished by ensuring that
+the operation can only be included once in a block and that it is always very light. In order to
+make it more difficult for malicious validators to abuse these types of dispatches they may not be
+included in blocks that return errors. This dispatch class exists to serve the assumption that it is
+better to allow an overweight block to be created than to not allow any block to be created at all.
 
-* `base_fee`: A fixed fee that is applied to every transaction. See
-  [`TransactionBaseFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionBaseFee).
-* `length_fee`: A per-byte fee that is multiplied by the length, in bytes, of the encoded
-  transaction. See
-  [`TransactionByteFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionByteFee).
-* `weight_fee`: A per-weight-unit fee that is multiplied by the weight of the transaction. The
-  weight of each dispatch is denoted via the flexible `#[weight]` annotation. Knowing the weight, it
-  must be converted to the `Currency` type. For this, each runtime must define a
-  [`WeightToFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.WeightToFee)
-  type that makes the conversion. `WeightToFee` must be a struct that implements [`Convert<Weight,
-  Balance>`](https://substrate.dev/rustdocs/master/sp_runtime/traits/trait.Convert.html).
+#### `FixedMandatory`
 
-Based on the above, the final fee of a dispatchable is:
+This function will have a fixed weight and be allowed in a block even if it makes it overweight.
 
-```
-fee =
-  base_fee +
-  len(tx) * length_fee +
-  WeightToFee(weight)
+```rust
+#[weight = SimpleDispatchInfo::FixedMandatory(20)]
+pub fn block_validation_critical_function() { block_validation_op(); }
 ```
 
-This `fee` is known as the "inclusion fee." Even if the extrinsic fails, the signer must pay this
-inclusion fee.
+> **Note:** Be careful! Functions that do not specify a dispatch type will get the default value of
+> `FixedNormal(10_000)`. This is due to how things work in `substrate-node` and the desired
+> granularity of Substrate. Even if you want to use the `SimpleDispatchInfo`, it is very likely that
+> you would want it to have a different default.
+
+## Fee Calculation
+
+The final fee of a dispatch is calculated using the inclusion fee and a number of configurable
+parameters.
```

Comment:
The final fee is the inclusion fee plus the (optional) tip. The inclusion fee varies _based on_ configurable parameters.